src/fasta-recs.py

```diff
@@ -11,7 +12,50 @@ def main():
     )
     args = argparser.parse_args()
 
-    print(f"Now I need to process the records in {args.fasta}")
+    # Read the content
+    content = args.fasta.read()
+    
+    index = 0
+
+    while(index<len(content)):
+        # Look for '> *'
+        match = re.search(">(\s)*", content[index:])
+        if match is None:
+            break
+        index += match.end()
+
+        beginName = index
+
+        # Look for the end of the name
+        match = re.search("\n", content[index:])
+        if match is None:
+            chain = content[beginName:len(content)]
+            chain = re.sub('(\s)+',' ',chain).strip()
```

Comment:
Each of these commands take time (at least, and in this case exactly) proportional to the length of the string you are modifying. This is not as much of a problem as the slicing above, because you extract something that you only process once, but later in the classs we run into cases where slicing to get a substring is a big no-no because of the linear running time. So keep that in mind. Again, it is not a problem here, I'm just pointing out the complexity of slicing in Python, and that you might want to keep it in mind. (In other languages, you can slice in constant time, but Python is not one of those).